src/baseline_automatic_pipeline_system/parse_okr_v2.py

```diff
@@ -0,0 +1,272 @@
+                                                    argument_mentions=argument_mentions,
+                                                    is_explicit=True)
+
+            # TODO this will also be modified afterwards
+            mention_object.template = mention["Template"]
+
+            # add this PropositionMention object to Proposition.mentions dict
+            prop_mentions[new_mention_id] = mention_object
+            # maintain global mapping
+            global_mention_id_to_mention_object[mention_global_id] = mention_object
+
+        # add this prop as Proposition object to okr.propositions
+        all_prop_terms = [ mention.terms for mention in prop_mentions.values() ]
+        okr_info["propositions"][prop_id] = okr.Proposition(id=prop_id,
+                                                            name=all_prop_terms[0],
+                                                            mentions=prop_mentions,
+                                                            attributor=None,  # TODO neccessary?
+                                                            terms=all_prop_terms,
+                                                            entailment_graph=None)
+
+    # modify PropositionMention - parent_mention_id, parent_id, and template
+    for prop_id, prop in okr_info["propositions"].items():
+        for prop_mention_id, prop_mention in prop.mentions.items():
+            # modify arguments
+            for argument_mention_id, argument_mention in prop_mention.argument_mentions.items():
+                mention_global_id = argument_mention.parent_mention_id
+                sent_id, symbol = mention_global_id.split("_")
+                # modify parent_mention_id
+                arg_orig_mention_object = global_mention_id_to_mention_object[mention_global_id]
+                argument_mention.parent_mention_id = arg_orig_mention_object.id
+                argument_mention.parent_id = arg_orig_mention_object.parent
+
+                # replace symbol of argument in template with id of its parent (Entity/Proposition)
+                prop_mention.template = prop_mention.template.replace("{"+symbol+"}", "{"+argument_mention.parent_id+"}")
+
+    return okr_info
+
+
+# main
+if __name__ == "__main__":
+    # general settings
+    logging.basicConfig(level=logging.INFO)
+
+    # Parse arguments
+    args = docopt(__doc__)
+    input_fn = args["--in"]
+    output_fn = args["--out"]
+
+    #automatic pipeline
+    sentences = get_raw_sentences_from_file(input_fn)
+    parsed_sentences = parse_single_sentences(sentences)
+    all_entity_mentions, all_proposition_mentions = get_mention_lists(parsed_sentences)
+    entities = cluster_entities(all_entity_mentions)
+    propositions = cluster_propositions(all_proposition_mentions)
+    okr_info = generate_okr_info(sentences, all_entity_mentions, all_proposition_mentions, entities, propositions)
+    okr_v1 = okr.OKR(**okr_info)
```

Comment:
Nit: why use an intermediate variable, instead of:
```python
okr_v1 = okr.OKR(**generate_okr_info(sentences, all_entity_mentions, all_proposition_mentions, entities, propositions))
```
Seems cleaner to me.